handlers/admin/datamanager/security_group.cfc

```diff
@@ -0,0 +1,130 @@
+				  link       = event.buildAdminLink( objectName=objectName, recordId=record.id )
+				, icon       = "fa-eye"
+				, contextKey = "v"
+			} );
+		}
+
+		if ( hasCmsPermission( "groupmanager.edit" ) ) {
+			ArrayAppend( actions, {
+				  link       = event.buildAdminLink( objectName=objectName, recordId=record.id, operation="editRecord", queryString="op=object" )
+				, icon       = "fa-pencil"
+				, contextKey = "e"
+			} );
+		}
+
+		if ( hasCmsPermission( "groupmanager.delete" ) && !isTrue( record.is_catch_all ?: "" ) ) {
+			ArrayAppend( actions, {
+				  link       = event.buildAdminLink( objectName=objectName, recordId=record.id, operation="deleteRecordAction" )
+				, icon       = "fa-trash"
+				, contextKey = "d"
+				, class      = "confirmation-prompt"
+				, title      = translateResource( uri="cms:datamanager.deleteRecord.prompt", data=[ translateResource( uri="preside-objects.#objectName#:title.singular", defaultValue=objectName ), record.label ] )
+				, match      = dataManagerService.useTypedConfirmationForDeletion( objectName ) ? datamanagerService.getDeletionConfirmationMatch( objectName, record ) : ""
+			} );
+		} else {
+			ArrayAppend( actions, {
+				  link       = "##"
+				, icon       = "fa-trash grey"
+				, contextKey = "d"
+			} );
+		}
+
+		return actions;
+	}
+
+	private string function renderSidebarHeader( event, rc, prc, args={} ) {
+		if ( !isEmptyString( args.record.id ?: "" ) ) {
+			return renderView( view="/admin/datamanager/security_group/_sidebarHeader", args=args );
+		}
+
+		return "";
+	}
+
+	private array function getTopRightButtonsForViewRecord( event, rc, prc, args ) {
+		var recordId    = args.recordId ?: "";
+		var recordLabel = args.record.known_as ?: "";
+
+		var actions  = [];
+		var children = [];
+
+		if ( prc.canDelete && !isTrue( args.record.is_catch_all ?: "" ) ) {
+			ArrayAppend( children, {
+				  link      = event.buildAdminLink( objectName="security_group", recordId=prc.recordId, operation="deleteRecordAction" )
+				, icon      = "fa-trash red"
+				, globalKey = "d"
+				, title     = translateResource( uri="cms:datamanager.deleteRecord.btn" )
+				, prompt    = translateResource( uri="cms:datamanager.deleteRecord.prompt", data=[ prc.objectTitle, stripTags( prc.recordLabel ) ] )
+				, match     = datamanagerService.getDeletionConfirmationMatch( "security_group", QueryRowToStruct( prc.record ) )
+			} );
+		}
+
+		if ( prc.canEdit ) {
+			ArrayAppend( actions, {
+				  link      = event.buildAdminLink( objectName="security_group", recordId=recordId, operation="editRecord" )
+				, btnClass  = "btn-primary-default"
+				, iconClass = "fa-pencil"
+				, globalKey = "e"
+				, title     = translateResource( "cms:datamanager.editRecord.btn" )
+				, children  = children
+			} );
+		}
+
+		return actions;
+	}
+
+	private void function preRenderAddRecordForm( event, rc, prc, args={} ) {
+		args.cancelAction = event.buildAdminLink( linkTo="adminManager.groups" );
```

Comment:
I think you can avoid most of these custom overrides of cancel/success URLs by implementing `buildListingLink()`:

```cfc
private function buildListingLink() {
    return event.buildAdminLink( linkto="adminManager.groups", queryString=args.queryString ?: "" );
}
```

Same applies to similar logic in `security_user.cfc`